python_modules/libraries/dagster-dbt/dagster_dbt_tests/test_dbt_python_dependencies.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+from dagster import (
+    AssetKey,
+    asset,
+    file_relative_path,
+)
+from dagster._core.errors import DagsterInvalidInvocationError
+from dagster_dbt.asset_decorator import dbt_assets
+from dagster_dbt.cli import DbtManifest
+
+test_dagster_metadata_manifest = DbtManifest.read(
+    path=file_relative_path(__file__, "dbt_projects/test_dagster_metadata/manifest.json")
+)
+
+
+def test_get_asset_key_for_dbt_unique_id() -> None:
+    assert test_dagster_metadata_manifest.get_asset_key_for_dbt_unique_id(
+        "source.test_dagster_metadata.jaffle_shop.raw_events"
+    ) == AssetKey(["jaffle_shop", "raw_events"])
+
+
+def test_get_explicit_asset_key_for_dbt_unique_id_() -> None:
+    assert test_dagster_metadata_manifest.get_asset_key_for_dbt_unique_id(
+        "source.test_dagster_metadata.jaffle_shop.raw_customers"
+    ) == AssetKey(["customized", "source", "jaffle_shop", "main", "raw_customers"])
+
+
+def test_asset_downstream_of_dbt_asset() -> None:
+    upstream_asset_key = AssetKey(["orders"])
+
+    @dbt_assets(manifest=test_dagster_metadata_manifest)
+    def my_dbt_assets():
+        ...
+
+    @asset(
+        non_argument_deps={
+            test_dagster_metadata_manifest.get_asset_key_for_dbt_unique_id(
```

Comment:
I assume users won't be calling `get_asset_key_for_dbt_unique_id` directly, so let's use `get_asset_key_for_dbt_model` here instead.